tds-prize-tool/src/rewards_earned.rs

```diff
@@ -0,0 +1,282 @@
+use crate::prize::{self, Winner, Winners};
+use solana_runtime::bank::Bank;
+use solana_sdk::account::Account;
+use solana_sdk::pubkey::Pubkey;
+use solana_stake_api::stake_state::StakeState;
+use solana_vote_api::vote_state::VoteState;
+use std::cmp::{max, min};
+use std::collections::HashMap;
+
+pub const STARTING_BALANCE_LAMPORTS: i64 = 0; // TODO
+
+const HIGH_BUCKET: &str = "Top 25% Bucket";
+const MID_BUCKET: &str = "Top 25-50% Bucket";
+const LOW_BUCKET: &str = "Top 50-90% Bucket";
+
+fn voter_stake_rewards(stake_accounts: HashMap<Pubkey, Account>) -> HashMap<Pubkey, u64> {
+    let mut voter_stake_sum: HashMap<Pubkey, u64> = HashMap::new();
+    for (_key, account) in stake_accounts {
+        if let Some(StakeState::Stake(_authorized, _lockup, stake)) = StakeState::from(&account) {
+            let stake_sum = voter_stake_sum
+                .get(&stake.voter_pubkey)
+                .cloned()
+                .unwrap_or_default();
+            voter_stake_sum.insert(stake.voter_pubkey, stake_sum + account.lamports);
+        }
+    }
+    voter_stake_sum
+}
+
+fn validator_results(validator_reward_map: HashMap<Pubkey, u64>) -> Vec<(Pubkey, i64)> {
+    let mut validator_rewards: Vec<(Pubkey, u64)> = validator_reward_map
+        .iter()
+        .map(|(key, balance)| (*key, *balance))
+        .collect();
+
+    // Sort descending and calculate results
+    validator_rewards.sort_by(|a, b| b.1.partial_cmp(&a.1).unwrap());
+    validator_rewards
+        .into_iter()
+        .map(|(key, earned)| (key, (earned as i64) - STARTING_BALANCE_LAMPORTS))
```

Comment:
might want to take STARTING_BALANCE_LAMPORTS as a parameter for this tool?

Reply:
@mvines are all validators starting with the same balance?

Reply:
yes, in that everybody starts with a system account with 1000 SOL in it.